cdap-ui/app/cdap/components/SortableTable/index.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {Component, PropTypes} from 'react';
+import orderBy from 'lodash/orderBy';
+import classnames from 'classnames';
+require('./SortableTable.scss');
+
+export default class SortableTable extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      entities: props.entities,
+      sortByHeader: '',
+      sortOrder: 'asc'
+    };
+
+    this.renderTableBody = this.props.renderTableBody.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let entities = nextProps.entities;
+    let sortByHeader = this.state.sortByHeader;
+    if (!sortByHeader) {
+      sortByHeader = this.getDefaultSortedHeader();
+    }
+    entities = orderBy(entities, [sortByHeader], [this.state.sortOrder]);
+    this.setState({
+      entities,
+      sortByHeader
+    });
+  }
+
+  getDefaultSortedHeader() {
+    let defaultHeader = '';
+    this.props.tableHeaders.every(header => {
+      if (header.property) {
+        defaultHeader = header.property;
+        return false;
+      }
+      return true;
+    });
```

Comment:
`.every` iterates through the array until it sees a falsy condition, so here I'm stopping the loop when I see a sortable header (i.e. a header attached to the value of a property). I agree it's not very readable though.